framework/src/service/ServiceReferenceBasePrivate.cpp

```diff
@@ -106,7 +117,7 @@ std::shared_ptr<void> ServiceReferenceBasePrivate::GetService(BundlePrivate* bun
   auto s = ExtractInterface(GetServiceInterfaceMap(bundle), interfaceId);
   if (!s)
   {
-    registration->Lock(), --registration->dependents[bundle];
+    registration->Lock(), --registration->dependents.at(bundle);
```

Comment:
For a valid `InterfaceMapConstPtr` the code looks fine, but I am concerned about the invalid case.

First, if `GetServiceInterfaceMap` returns a `null` map (in case `registration->available` if `false` or due to some other condition (e.g. the race you found), then `ExtractInterface` will dereference an nullptr (we should fix that).

Second, in the invalid case, the counter was not incremented in `GetServiceInterfaceMap` (I believe) but we decrement it here. The decrement is necessary if `GetServiceInterfaceMap` returns a valid map (and hence incremented the counter) but `ExtractInterface` fails (because e.g. `interfaceId` is empty. We currently do not differentiate the two cases.